core/src/main/kotlin/org/evomaster/core/search/gene/ObjectGene.kt

```diff
@@ -42,6 +44,26 @@ open class ObjectGene(name: String, val fields: List<out Gene>, val refType: Str
         }
     }
 
+    fun copyFields(filterGene: ObjectGene): ObjectGene {
+
+        val fields: MutableList<Gene> = mutableListOf()
+        var k = 0
+        for (i in this.fields.indices) {
+            var exist = false
+            for (element in filterGene.fields) {
+                if (this.fields[i].name == element.name) {
+                    exist = true
+                    break
+                }
+            }
+            if (!exist) {
+                fields.add(this.fields[i].copy())
+                k++
+            }
+        }
+        return ObjectGene(this.name, fields)
+    }
```

Comment:
Hi, @arcuri82 I have checked it with @asmab89 , and added a test for it. Depending on where `copyField` is invoked, it might miss the binding references. Now as I checked, it is used for creating genes, it would be ok.

Reply:
Comment added.